examples/graphiql-plugin.js

```diff
@@ -0,0 +1,115 @@
+    setExplain(sampleData) { this.sampleData = sampleData || []; this.dispatchEvent(new Event('updateSampleData')); }
+  }
+
+  const sampleDataManager = new SampleDataManager();
+  const useSampleData = () => {
+    const [sampleData, setSampleData] = React.useState(sampleDataManager.getExplainData());
+    React.useEffect(() => {
+      const eventListener = sampleDataManager.addEventListener(
+        'updateSampleData', (e, value) => {setSampleData(_ => e.target.sampleData || []);});
+      return () => {sampleDataManager.removeEventListener('updateSampleData', eventListener);}
+    }, []);
+    return {sampleData}
+  };
+
+  function Content() {
+    const { sampleData } = useSampleData();
+    return (
+      React__default["default"].createElement('div', {style: { maxWidth: '300px'}}, [
+        React__default["default"].createElement('div', {style: { height: '100%'}}, ["This is a sample plugin"]),
+        sampleData && React__default["default"].createElement('pre', null, [JSON.stringify(sampleData, null, 2)])
+      ]))}
+
+  function Icon() {return React__default["default"].createElement('p', {style: { width: '100%', display: 'flex', justifyContent: 'center'}}, ["P"])}
+  function graphiqlSamplePlugin() {return umdPlugin()}
+  
+  /*
+   * The plugin should have \`umdPlugin\` function that exports:
+   *   title: string. The title of the plugin
+   *   icon: React component. The icon shown in the toolbar
+   *   content: React component.
+   */  
+  function umdPlugin() {
+    return { 
+      title: 'GraphiQL Sample',
+      icon: () => Icon(),
+      content: () => { return Content() }
+    }
+  }
+
+  /*
+   * A function to intercept the fetch data can be exported to allow access to the data.
+   * Graphiql do not provide an easy way to access the retrieved data.
+   */  
+  function parseFetchResponse(data) {
+    if (data.data) {sampleDataManager.setExplain(data.data);} // Do something with the data
+    return data // It should return the data
+  }
+  exports.graphiqlSamplePlugin = graphiqlSamplePlugin;
+  exports.parseFetchResponse = parseFetchResponse;
+  exports.umdPlugin = umdPlugin;
+  Object.defineProperty(exports, '__esModule', { value: true });
+}));
+`
+
+const schema = `
+  type Query {
+    add(x: Int, y: Int): Int
+  }
+`
+
+const resolvers = {
+  Query: {
+    add: async (_, obj) => {
+      const { x, y } = obj
+      return x + y
+    }
+  }
+}
+
+app.register(mercurius, {
+  schema,
+  resolvers,
+  graphiql: {
+    enabled: true,
+    plugins: [
+      {
+        name: 'samplePlugin',
+        props: {},
+        umdUrl: 'http://localhost:3000/graphiql/sample.js',
+        fetcherWrapper: 'parseFetchResponse'
+      }
+    ]
+  }
+})
+
+app.get('/', async function (req, reply) {
+  const query = '{ add(x: 2, y: 2) }'
+  return reply.graphql(query)
+})
+
+app.get('/graphiql/sample.js', (req, reply) => {
+  reply
+    .header('Content-Type', 'application/javascript')
+    .send(pluginSrc)
```

Comment:
this is different from the example, which uses sendFile. I would try to be more consistent

Reply:
Now that I've a folder the plugin is in a separated file, `sendFile` is now used.